crates/flux-rustc-bridge/src/mir.rs

```diff
@@ -750,7 +750,7 @@ impl fmt::Debug for Constant {
             Constant::Bool(b) => write!(f, "{b}"),
             Constant::Unit => write!(f, "()"),
             Constant::Str(s) => write!(f, "\"{s:?}\""),
-            Constant::Char => write!(f, "\"<opaque char>\""),
+            Constant::Char(c) => write!(f, "\'{}\'", unsafe { char::from_u32_unchecked(*c) }),
```

Comment:
This becomes 

```suggestion
            Constant::Char(c) => write!(f, "\'{c}\'"),
```

A note on the way you used `unsafe` for educational purposes because it won't matter after we store a `char` instead of `u32`: The `Constant::Char` constructor is public, so anyone can call it passing an arbitrary integer. This means you *cannot* assume the integer is a valid char inside unsafe code.